faq/osgi/README.md

```diff
@@ -0,0 +1,179 @@
+## Frequently Asked Questions - OSGi
+
+### How to use the Java driver in an OSGi environment?
+
+We have complete examples demonstrating usage of the driver in an [OSGi]
+environment; please refer to our [OSGi examples repository].
+
+
+### How to override Guava's version?
+
+The driver is compatible and tested with all versions of Guava in the range
+`[16.0.1,20)`.
+
+If using Maven, you can force a more specific version by re-declaring
+the Guava dependency in your project, e.g.:
+
+    <dependency>
+        <groupId>com.google.guava</groupId>
+        <artifactId>guava</artifactId>
+        <version>19.0</version>
+    </dependency>
+
+Make sure that your project's manifest is importing the right version
+of Guava's packages, e.g. for 19.0:
+
+    Import-Package: com.google.common.base;version="[19.0,20)"
+
+
+### How to enable compression?
+
+First, read our [manual page on compression](../../manual/compression/)
+to understand how to enable compression for the Java driver.
+            
+OSGi projects can use both Snappy or LZ4 compression algorithms. 
+
+For Snappy, include the following Maven dependency:
+
+    <dependency>
+        <groupId>org.xerial.snappy</groupId>
+        <artifactId>snappy-java</artifactId>
+        <version>1.1.2.6</version>
+    </dependency>
+
+For LZ4, include the following Maven dependency:
+
+    <dependency>
+        <groupId>net.jpountz.lz4</groupId>
+        <artifactId>lz4</artifactId>
+        <version>1.3.0</version>
+    </dependency>
+
+**IMPORTANT**: versions of LZ4 library below 1.3.0 cannot be used
+because they are _not_ valid OSGi bundles.
+
+Because compression libraries are _optional runtime dependencies_, 
+most manifest generation tools (such as [BND] and the [Maven bundle plugin]) 
+will not reference these libraries in your project's manifest.
+This is correct, but could be a problem for some OSGi provisioning tools,
+and notably for [Tycho], because it does not consider optional
+dependencies when computing the target platform.
+If you are facing provisioning issues related to compression libraries,
+you might need to reference them in your project's manifest explicitly; 
+with the [Maven bundle plugin], this can be achieved with the following 
+[BND] `Import-Package` instruction (the example below is for
+LZ4, but the same applies to Snappy as well):
+
+    <Import-Package>net.jpountz.lz4,*</Import-Package>
```

Comment:
Hmm you're right, the syntax highlighting on Github seems confused by this asterisk, but I checked the grammar and inside a code block like this, there should be no parsing, so I guess the problem is with Github. Anyway the rendered HTML looks fine.